overlord/snapshotstate/backend/backend.go

```diff
@@ -321,6 +323,11 @@ func Save(ctx context.Context, id uint64, si *snap.Info, cfg map[string]interfac
 		// Note: Auto is no longer set in the Snapshot.
 	}
 
+	archiveOptions := &addDirToZipOptions{}
+	if err := readSnapshotYaml(si, archiveOptions); err != nil {
```

Comment:
Between the two variant I prefer this one, because it makes the code in `readSnapshotYaml` a bit cleaner (just one return value instead of two) and removes the need for nil checks in the caller. But I don't have a strong preference, so I'll bend to the majority :-)

Reply:
If it returned a struct value instead of a pointer, the caller wouldn't need nil checks I think. I see your point though. I guess a return value just seemed more "natural" than an output parameter but it's also not a strong preference so feel free to ignore

Reply:
I tweaked this